main.tf

```diff
@@ -8,11 +8,11 @@ module "ssh-key" {
 }
 
 resource "azurerm_kubernetes_cluster" "main" {
-  name                    = "${var.prefix}-aks"
+  name                    = var.cluster_name == "" ? "${var.prefix}-aks" : var.cluster_name
```

Comment:
It should be `var.cluster_name == null ? "${var.prefix}-aks" : var.cluster_name`. 
If `cluster_name` not defined, it returns error in you code.